clang/lib/CodeGen/ItaniumCXXABI.cpp

```diff
@@ -4907,7 +4923,8 @@ void XLCXXABI::registerGlobalDtor(CodeGenFunction &CGF, const VarDecl &D,
   }
 
   // Create __dtor function for the var decl.
-  llvm::Function *DtorStub = CGF.createAtExitStub(D, Dtor, Addr);
+  llvm::Function *DtorStub =
+      cast<llvm::Function>(CGF.createAtExitStub(D, Dtor, Addr));
```

Comment:
It looks like here we have the same issue as @asl already commented on above in CGDeclCXX.cpp.

Reply:
Here it is used for XLC ABI which seems to be never using pauth as far as I can see :)